ros/2/src/openvslam/src/openvslam_ros.cc

```diff
@@ -53,15 +53,50 @@ mono::mono(const std::shared_ptr<openvslam::config>& cfg, const std::string& voc
         node_.get(), "camera/image_raw", [this](const sensor_msgs::msg::Image::ConstSharedPtr& msg) { callback(msg); }, "raw", custom_qos_);
 }
 void mono::callback(const sensor_msgs::msg::Image::ConstSharedPtr& msg) {
-    const auto tp_1 = std::chrono::steady_clock::now();
-    const auto timestamp = std::chrono::duration_cast<std::chrono::duration<double>>(tp_1 - tp_0_).count();
-
+    const rclcpp::Time tp_1 = node_->now();
+    const double timestamp = (tp_1 - tp_0_).nanoseconds();
+    
     // input the current frame and estimate the camera pose
     SLAM_.feed_monocular_frame(cv_bridge::toCvShare(msg)->image, timestamp, mask_);
```

Comment:
Is there a reason these can't be absolute values of time and not time deltas (e.g. why even bother with `tp_0_`)?

Reply:
Well, the slam system ran fine using timestamp = tp_1, and even when equal to a constant... I need to look further where this timestamp is used internally for a better understand...

Reply:
The correct unit of time is seconds, not nanoseconds.